src/adam_core/propagator/adam_assist.py

```diff
@@ -292,11 +292,16 @@ def _detect_impacts(
         sim = None
         sim = rebound.Simulation()
 
+        backward_propagation = num_days < 0
+
         # Set the simulation time, relative to the jd_ref
         start_tdb_time = orbits.coordinates.time.jd().to_numpy()[0]
         start_tdb_time = start_tdb_time - ephem.jd_ref
         sim.t = start_tdb_time
 
+        if backward_propagation:
+            sim.dt = -0.01
```

Comment:
Can we change this to simply multiply the existing sim.dt by -1?
```suggestion
            sim.dt = sim.dt * -1
```